lib/src/test/kotlin/starknet/provider/ProviderTest.kt

```diff
@@ -424,73 +498,163 @@ class ProviderTest {
         assertTrue(response is GatewayTransactionReceipt)
     }
 
-    // FIXME this test will fail until devnet spec is updated as there is no way to differentiate between declare
-    //  and deploy tx receipts currently
-//    @Test
-//    fun `get deploy transaction receipt rpc`() {
-//        val request = rpcProvider().getTransactionReceipt(deployTransactionHash)
-//        val response = request.send()
-//
-//        assertNotNull(response)
-//        assertTrue(response is DeployTransactionReceipt)
-//    }
-
     @Test
-    fun `get declare transaction receipt rpc`() {
-        val request = rpcProvider().getTransactionReceipt(declareTransactionHash)
+    fun `get l1 handler transaction receipt rpc`() {
+        val httpService = mock<HttpService> {
+            on { send(any()) } doReturn HttpResponse(
+                true,
+                200,
+                """
```

Comment:
How do I fetch this tx myself? Please make sure you add info about this to every mock.

Reply:
I have added info to all the gateway transactions, but with rpc it's trickier (as you have to have an rpc node). Should I add the commands even though that's the case? or gateway commands suffice?